src/query_ctx.h

```diff
@@ -70,6 +72,8 @@ void QueryCtx_SetGraphCtx(GraphContext *gc);
 void QueryCtx_SetResultSet(ResultSet *result_set);
 /* Set the last writer which needs to commit */
 void QueryCtx_SetLastWriter(OpBase *op);
+/* Set the query's associated timeout job. */
+void QueryCtx_SetTimeoutJob(CronTask task);
```

Comment:
See if it make sense to change this to: `QueryCtx_SetTimeout(ms,... )`

Reply:
I don't think that would work easily - the `QueryCtx` tracks whether the query has an associated timeout, it does not instantiate it.